mlc_llm/utils.py

```diff
@@ -108,17 +108,48 @@ def split_transform_deploy_mod(
 
 
 def transform_params(
-    mod_transform: tvm.IRModule, model_params: List[tvm.nd.NDArray]
+    mod_transform: tvm.IRModule,
+    model_params: List[tvm.nd.NDArray],
 ) -> List[tvm.nd.NDArray]:
+    # Remove the dataflow block inside the param transform function,
+    # so that the LazyTransformParams pass can be applied.
+    mod_transform = relax.transform.ToNonDataflow()(mod_transform)
+    mod_transform = relax.transform.LazyTransformParams()(mod_transform)
+
     transform_func_name = None
     for gv, func in mod_transform.functions.items():
         if isinstance(func, relax.Function):
             transform_func_name = gv.name_hint
     assert transform_func_name is not None
 
-    ex = relax.build(mod_transform, target="llvm")
-    vm = relax.vm.VirtualMachine(ex, tvm.cpu())
-    res = vm[transform_func_name](model_params)
+    if tvm.cuda().exist:
+        target = "cuda"
+    elif tvm.metal().exist:
+        target = "metal"
+    else:
+        target = "llvm"
+    target = tvm.target.Target(target)
+    device = tvm.device(target.kind.default_keys[0])
+
+    @tvm.register_func("get_item")
```

Comment:
maybe `allow_override=True`?

Reply:
I think `get_item` and `set_item` are too general, we can rename them to `lazy_transform.set_item` and `lazy_transform.get_item` instead.
However, it involves changes on relax side, we can do it in later PRs.

Reply:
Agreed. This required name should be updated with a more specific one for better clarity.

Reply:
Updated with `override=True` @junrushao.